### PR TITLE
vk: add Metal capture support using MoltenVK, part 1

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**

### CODE REVIEW

File: libnodegl/gpu_capture_metal.m [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+int gpu_capture_init(struct gpu_capture_ctx *s)
+{
+    return 0;
+}
+
+int gpu_capture_begin(struct gpu_capture_ctx *s)
+{
+    NSError *error;
+    MTLCaptureManager *captureManager = [MTLCaptureManager sharedCaptureManager];
+    MTLCaptureDescriptor *captureDescriptor = [[MTLCaptureDescriptor alloc] init];
```
Comment (mbouron-gpsw) [inline libnodegl/gpu_capture_metal.m:54]: Can you check that the object has properly be initialized ? ie:
```c
if (!captureDescriptor)
    return NGL_ERROR_EXTERNAL;
```

File: libnodegl/gpu_capture_metal.m [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+int gpu_capture_init(struct gpu_capture_ctx *s)
+{
+    return 0;
+}
+
+int gpu_capture_begin(struct gpu_capture_ctx *s)
+{
+    NSError *error;
+    MTLCaptureManager *captureManager = [MTLCaptureManager sharedCaptureManager];
```
Comment (mbouron-gpsw) [inline libnodegl/gpu_capture_metal.m:53]: Can you use the snake case convention everywhere ?

File: libnodegl/gpu_capture_metal.m [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        struct gctx_vk *gctx_vk = (struct gctx_vk *)s->gctx;
+        struct vkcontext *vkcontext = gctx_vk->vkcontext;
+        id<MTLDevice> mtl_device;
+        vkGetMTLDeviceMVK(vkcontext->phy_device, &mtl_device);
+        captureDescriptor.captureObject = mtl_device;
+    }
+#endif
+    captureDescriptor.destination = MTLCaptureDestinationDeveloperTools;
+    if (![captureManager startCaptureWithDescriptor: captureDescriptor error:&error]) {
+        NSLog(@"Failed to start capture, error %@", error);
```
Comment (mbouron-gpsw) [inline libnodegl/gpu_capture_metal.m:66]: Can you use `LOG(ERROR, "failed to start capture, error: %s", [error.description UTF8String]);` instead ?

File: libnodegl/gpu_capture_metal.m (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/gpu_capture_metal.m]: updated

File: libnodegl/gpu_capture_metal.m (hunk 2 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/gpu_capture_metal.m]: updated

File: libnodegl/gpu_capture_metal.m (hunk 3 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/gpu_capture_metal.m]: updated

**Review by mbouron-gpsw — APPROVED** [review]
LGTM.

**Pull request merged**